c_src/adbc_nif.cpp

```diff
@@ -1459,7 +1459,7 @@ static int on_load(ErlNifEnv *env, void **, ERL_NIF_TERM) {
 
     {
         using res_type = NifRes<struct AdbcStatement>;
-        rt = enif_open_resource_type(env, "Elixir.Adbc.Nif", "NifResAdbcStatement", res_type::destruct_resource, ERL_NIF_RT_CREATE, NULL);
+        rt = enif_open_resource_type(env, "Elixir.Adbc.Nif", "NifResAdbcStatement", destruct_adbc_statement_resource, ERL_NIF_RT_CREATE, NULL);
```

Comment:
Ah, this is much better! Shall we also define explicit callbacks for AdbcConnection, error, and arrrow array stream? I think explicit callbacks will be cleaner than meta-programming/template. :)